### PR TITLE
Accounting export apis added

### PR DESCRIPTION
**Pull request by ruuushhh**

### CODE REVIEW

File: apps/accounting_exports/models.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    Table to store accounting exports
+    """
+    id = models.AutoField(primary_key=True)
+    type = StringNotNullField(max_length=50, help_text='Task type (FETCH_EXPENSES / INVOICES / DIRECT_COST)')
+    fund_source = StringNotNullField(help_text='Expense fund source')
+    mapping_errors = ArrayField()
+    expenses = models.ManyToManyField(Expense, help_text="Expenses under this Expense Group")
+    task_id = StringNullField(help_text='Fyle Jobs task reference')
+    description = CustomJsonField(help_text='Description')
+    invoice = models.ForeignKey(Invoice, on_delete=models.PROTECT, help_text='Referenc to Invoie', null=True)
```
Comment (Shwetabhk) [inline apps/accounting_exports/models.py:26]: move this to Invoice Table

File: apps/accounting_exports/models.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    direct_cost = models.ForeignKey(DirectCost, on_delete=models.PROTECT, help_text='Reference to DirectCost', null=True)
```
Comment (Shwetabhk) [inline apps/accounting_exports/models.py:27]: Move this to Direct Cost

File: apps/fyle/models.py [hunk 3]
```diff
@@ -53,6 +58,49 @@ class Meta:
         db_table = 'expense_filters'
 
 
+class Expense(models.Model):
```
Comment (Shwetabhk) [inline apps/fyle/models.py:61]: Add accounting_export_summary columns

File: apps/sage300/models.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    id = models.AutoField(primary_key=True)
+    amount = FloatNullField(help_text='Invoice amount')
+    code = StringNotNullField(help_text='Code Id')
+    job_id = StringNotNullField(help_text='Job Id')
+    cost_code_id = StringNotNullField(help_text='Cost Code Id')
+    category_id = StringNotNullField(help_text='Category Id')
+    credit_card_account_id = StringNotNullField(help_text='Credit Card Account Id')
+    debit_card_account_id = StringNotNullField(help_text='Debit Card Account Id')
+    transaction_date = CustomDateTimeField(help_text='Transaction Date')
+    description = models.TextField(null=True, default='')
```
Comment (Shwetabhk) [inline apps/sage300/models.py:65]: use a custom field

File: apps/sage300/models.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+    id = models.AutoField(primary_key=True)
+    amount = FloatNullField(help_text='Invoice amount')
+    accounting_date = StringNotNullField(help_text='Accounting date')
+    description = models.TextField(null=True, default='')
+    tax_amount = FloatNullField(help_text='Tax amount')
+    vendor_id = StringNotNullField(help_text='Vendor ID')
+    code = StringNotNullField(max_length=15, help_text="unique key for each document")
+    discount_amount = FloatNullField(help_text='Discount amount')
+    workspace = models.ForeignKey(
```
Comment (Shwetabhk) [inline apps/sage300/models.py:34]: remove

File: apps/accounting_exports/serializers.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        status = self.context['request'].parser_context.get('query_params').get('status', None)
+
+        id = self.context['request'].parser_context.get('query_params').getlist('id')
+
+        start_date = self.context['request'].parser_context.get('query_params').get('start_date', None)
+
+        end_date = self.context['request'].parser_context.get('query_params').get('end_date', None)
+
+
+        filters = {
```
Comment (Shwetabhk) [inline apps/accounting_exports/serializers.py:30]: use django filters

File: apps/accounting_exports/models.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+
+class AccountingExport(BaseModel):
+    """
+    Table to store accounting exports
+    """
+    id = models.AutoField(primary_key=True)
+    type = StringNotNullField(max_length=50, help_text='Task type (FETCH_EXPENSES / INVOICES / DIRECT_COST)')
+    fund_source = StringNotNullField(help_text='Expense fund source')
+    mapping_errors = ArrayField()
```
Comment (NileshPant1999) [inline apps/accounting_exports/models.py:22]: why are we not adding help text here ??

File: apps/accounting_exports/models.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    id = models.AutoField(primary_key=True)
+    type = StringNotNullField(max_length=50, help_text='Task type (FETCH_EXPENSES / INVOICES / DIRECT_COST)')
+    fund_source = StringNotNullField(help_text='Expense fund source')
+    mapping_errors = ArrayField(help_text='Mapping errors', base_field=models.CharField(max_length=255), blank=True, null=True)
+    expenses = models.ManyToManyField(Expense, help_text="Expenses under this Expense Group")
+    task_id = StringNullField(help_text='Fyle Jobs task reference')
+    description = CustomJsonField(help_text='Description')
+    status = StringNotNullField(help_text='Task Status')
+    detail = CustomJsonField(help_text='Task Response')
+    sage_intacct_errors = CustomJsonField(help_text='Sage Intacct Errors')
```
Comment (NileshPant1999) [inline apps/accounting_exports/models.py:26]: its not intacct

File: apps/sage300/models.py [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+
+    id = models.AutoField(primary_key=True)
+    amount = FloatNullField(help_text='Invoice amount')
+    accounting_date = StringNotNullField(help_text='Accounting date')
+    description = TextNotNullField(help_text='Invoice description')
+    tax_amount = FloatNullField(help_text='Tax amount')
+    accounting_export = models.OneToOneField(AccountingExport, on_delete=models.PROTECT, help_text='Reference to AccountingExport model')
+    vendor_id = StringNotNullField(help_text='Vendor ID')
+    code = StringNotNullField(max_length=15, help_text="unique key for each document")
+    discount_amount = FloatNullField(help_text='Discount amount')
```
Comment (NileshPant1999) [inline apps/sage300/models.py:34]: i don't think we need this, as of now

File: apps/sage300/models.py [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+    amount: 12.31,
+    accounts_payable_account_id: '123123',
+    expense_account_id: '1231231',
+    description: 'Reimbursable Expenses by Shwetabh',
+    job_id: '123123',
+    cost_code_id: '12312123123'
+    category_id: '123'
+    """
+
+    id = models.AutoField(primary_key=True)
```
Comment (NileshPant1999) [inline apps/sage300/models.py:54]: lets do BIG auto field

File: apps/sage300/models.py [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
```
Comment (NileshPant1999) [inline apps/sage300/models.py:82]: same big auto field here

File: apps/accounting_exports/models.py [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+from apps.workspaces.models import BaseForeignWorkspaceModel
+from apps.fyle.models import Expense
+
+
+class AccountingExport(BaseForeignWorkspaceModel):
+    """
+    Table to store accounting exports
+    """
+    id = models.AutoField(primary_key=True)
+    type = StringNotNullField(max_length=50, help_text='Task type (FETCH_EXPENSES / INVOICES / DIRECT_COST)')
```
Comment (NileshPant1999) [inline apps/accounting_exports/models.py:18]: can we add it as an option field instead of char  ??

Comment (Shwetabhk) [reply apps/accounting_exports/models.py:18]: +1

File: apps/accounting_exports/models.py (hunk 8 above)
Comment (Shwetabhk) [reply apps/accounting_exports/models.py:26]: +1

File: apps/sage300/models.py (hunk 9 above)
Comment (Shwetabhk) [reply apps/sage300/models.py:34]: yeah lets remove this

**Review by Shwetabhk — APPROVED** [review]

File: tests/test_fyle/fixtures.py [hunk 13]
```diff
@@ -59,6 +59,45 @@
...
+    'accounting_export_response': {
+        "count":2,
+        "next":"None",
+        "previous":"None",
+        "results":[
+            {
+                "id":2,
+                "created_at":"2023-10-26T03:24:43.513291Z",
+                "updated_at":"2023-10-26T03:24:43.513296Z",
+                "type":"FETCHING_CREDIT_CARD_EXPENSES",
```
Comment (NileshPant1999) [inline tests/test_fyle/fixtures.py:71]: this is wrong type no ??

File: tests/conftest.py [hunk 14]
```diff
@@ -193,3 +194,30 @@ def add_expense_filters():
...
+            workspace_id=workspace_id,
+            type='FETCHING_REIMBURSABLE_EXPENSES',
+            defaults={
+                'status': 'IN_PROGRESS'
+            }
+        )
+
+        AccountingExport.objects.update_or_create(
+            workspace_id=workspace_id,
+            type='FETCHING_CREDIT_CARD_EXPENSES',
```
Comment (NileshPant1999) [inline tests/conftest.py:219]: this is not mentioned in the types

File: tests/conftest.py [hunk 15]
```diff
@@ -193,3 +194,30 @@ def add_expense_filters():
...
+    """
+    Pytest fixture to add accounting export to a workspace
+    """
+    workspace_ids = [
+        1, 2, 3
+    ]
+    for workspace_id in workspace_ids:
+        AccountingExport.objects.update_or_create(
+            workspace_id=workspace_id,
+            type='FETCHING_REIMBURSABLE_EXPENSES',
```
Comment (NileshPant1999) [inline tests/conftest.py:211]: this is also not added as option in the tupe

**Pull request merged**